libexec/ramalama/ramalama-serve-core

```diff
@@ -1,9 +1,104 @@
 #!/usr/bin/env python3
 
 import argparse
+import http.server
+import json
+import socketserver
 import sys
+import urllib.request
+from urllib.parse import urlparse
 
 
+class ProxyHandler(http.server.SimpleHTTPRequestHandler):
+    def log_message(self, format, *args):
+        # Override to prevent default logging
+        return
+
+    def do_GET(self):
+        self.copy_request_to_target()
+
+    def do_POST(self):
+        self.copy_request_to_target()
+
+    def do_PUT(self):
+        self.copy_request_to_target()
+
+    def do_DELETE(self):
+        self.copy_request_to_target()
+
+    def copy_request_to_target(self):
+        target_url = self.server.target_url + self.path
+        method = self.command
+        headers = self.headers
+        data = None
+
+        if 'Content-Length' in headers:
+            length = int(headers['Content-Length'])
+            data = self.rfile.read(length)
+
+        # Print request details
+        self.print_request_details(target_url, method, headers, data)
+
+        request = urllib.request.Request(target_url, data=data, headers=dict(headers), method=method)
+        
+        try:
+            with urllib.request.urlopen(request) as response:
+                self.send_response(response.status)
+
+                hop_by_hop_headers = ['connection', 'keep-alive', 'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'transfer-encoding', 'upgrade']
+                for key, value in response.getheaders():
+                    if key.lower() in hop_by_hop_headers:
+                        continue
+
+                    self.send_header(key, value)
+
+                self.end_headers()
+                response_data = ""
+                for line in response:
+                    self.wfile.write(line)
+                    self.wfile.flush()
+                    response_data += line.decode("utf-8")
+
+                # Print response details
+                self.print_response_details(response, response_data)
```

Comment:
Is this debugging data? Or at least something that should be hidden with args.quiet?

Reply:
Possibly the otherway around even, with a verbose option

Reply:
I don't really see much different between --debug and --verbose.

Reply:
Sorry yes, I mean vs args.quiet as in the first comment